### PR TITLE
B-20012 INT - GBLOC Switching for HQ User

### PR DESCRIPTION
**Pull request by ajlusk**
## Summary

Gives the HQ user the ability to switch GBLOCs so they can view the queues from all GBLOCs.

Adds a nullable query parameter to all queue endpoints that can specify a desired GBLOC to view as. If the requesting user is not an HQ user the parameter is ignored. The front end should never use this parameter for any non HQ user but I chose to ignore it in the back end to ensure access controls are still applied and to help prevent FE errors if somehow a non HQ user makes a request that includes this new parameter. 

Since the component that allows the user to select a GBLOC and the TableQueue that initiates the request are far apart in the component tree React Context was used to share state across distant components. [These docs](https://react.dev/learn/passing-data-deeply-with-context) and [this SO answer](https://stackoverflow.com/a/72758141) explains more about the approach used.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test
This affects a lot of places so this testing should be a lot of clicking around, and maybe trying to break it, to check nothing goes wrong.

1. Ensure the HQ Role FF is enabled.
2. Access the office app.
3. Login as an HQ user in any GBLOC.
4. In the header where the user's GBLOC is usually listed verify you have access to a drop down.
5. Select various GBLOCs from that dropdown and tab through the various queues ensuring everything displays correctly and updates when you select a new GBLOC or apply a filter or sort.
6. Spot check a couple selected GBLOC and selected Queue combinations with a non HQ user that is created in that GBLOC.
7. In Swagger, attempt to request each of the queues: Counseling (normal and PPM use the same endpoint), Moves, and Payment Requests while logged in as their usual user type (NOT HQ) with the viewAsGBLOC parameter set to a GBLOC other than that of the user you're logged in as. Verify the return value is for the logged in user's GBLOC rather than the requested GBLOC.

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2024-06-28 at 10 48 27 AM
Screenshot 2024-06-28 at 10 49 08 AM

### CODE REVIEW

**Comment by ajlusk** [comment]
Still needs some test fixes, new tests, and storybook but has been marked ready for review so the functional code can be reviewed and tested.

File: src/constants/queues.js
Comment (deandreJones) [inline src/constants/queues.js:1]: wondering if this list of gblocs should come from the DB- transportation_offices -distint gblocs

Comment (ajlusk) [reply src/constants/queues.js:1]: I like that idea. I was surprised I didn't find an existing list of all GBLOCs in the front-end. I copied this list from a different hard-coded list in swagger-def/ghc.yaml.

It's a fairly large addition (might warrant it's own ticket) but adding an endpoint for that seems like something that could come in handy elsewhere plus it's Sprint 5. I'll look into adding that.

File: swagger-def/ghc.yaml [hunk 1]
```diff
@@ -3446,6 +3452,12 @@ paths:
           name: orderType
           type: string
           description: order type
+        - in: query
+          name: viewAsGBLOC
+          type: string
+          description: |
+            Used to return a queue for a GBLOC other than the default of the current user. Requires the HQ role.
+            The parameter is ignored if the requesting user does not have the necessary role..
```
Comment (antgmann) [inline swagger-def/ghc.yaml:3460]: Super nitpick, but extra . on 3460

**Review by antgmann — APPROVED** [review]
Looks good, haven't been able to break it yet!

File: src/constants/queues.js
Comment (ajlusk) [reply src/constants/queues.js:1]: Discussed this idea with the team. We're all in favor of adding it but agreed it should be a separate enhancement backlog.

File: swagger-def/ghc.yaml (hunk 1 above)
Comment (ajlusk) [reply swagger-def/ghc.yaml]: Fixed 👍

**Review by r-mettler — APPROVED** [review]

**Pull request merged**